clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -189,16 +189,18 @@ ClientRequest newClientRequest(String nodeId, AbstractRequest.Builder<?> request
      *                         cancelling the request. The request may get cancelled sooner if the socket disconnects
      *                         for any reason including if another pending request to the same node timed out first.
      * @param callback the callback to invoke when we get a response
+     * @param initialPrincipalName the initial client principal name, when building a forward request
+     * @param initialClientId the initial client id, when building a forward request
      */
     ClientRequest newClientRequest(String nodeId,
                                    AbstractRequest.Builder<?> requestBuilder,
                                    long createdTimeMs,
                                    boolean expectResponse,
                                    int requestTimeoutMs,
+                                   String initialPrincipalName,
```

Comment:
can we have an overload for this that doesn't have these two fields, and defaults them to null?  It would really help avoid having so much churn in junit tests.

Reply:
I'm not sure it's meaningful to add an overload function just for the sake of minimizing code changes. Plus, I'm not sure whether this counts as a public interface, although it's not included in the standard java docs:
```
What are the "public interfaces" of the project?

- Binary log format
- The network protocol and api behavior
- Any class for which the build generates Javadoc.  
```

Reply:
Fair enough